test/grisp_connect_test_server.erl

```diff
@@ -198,8 +199,20 @@ wait_disconnection() ->
 
 %--- Websocket Callbacks -------------------------------------------------------
 
-init(Req, State) ->
-    {cowboy_websocket, Req, State}.
+init(Req, Opts) ->
+    ExpVer = maps:get(expected_protocol, Opts, <<"grisp-io-v1">>),
+    SelVer = maps:get(selected_protocol, Opts, <<"grisp-io-v1">>),
+    case cowboy_req:header(<<"sec-websocket-protocol">>, Req) of
+        undefined ->
+            Req2 = cowboy_req:reply(400, #{}, <<"No protocol specified">>, Req),
```

Comment:
I'm not user `400` is the correct way to respond here.

Reply:
Maybe we should set `sec-websocket-protocol: null` as below.

Reply:
As I said before, I think that if the server knows the handshake fail, it shouldn't rely on the client to reject it. 400 is maybe not the best status code though...

Reply:
Oh that looks interesting. But then we can also just leave it and use 400.